### ISSUE TITLE
JetBrains reproducible Crash using inline edits

### ISSUE THREAD

**Issue by Hakenadu** [continuedev/continue#1954]
### Before submitting your bug report

- [x] I believe this is a bug. I'll try to join the [Continue Discord](https://discord.gg/NWtdYexhMs) for questions
- [X] I'm not able to find an [open issue](https://github.com/continuedev/continue/issues?q=is%3Aopen+is%3Aissue) that reports the same bug
- [X] I've seen the [troubleshooting guide](https://docs.continue.dev/troubleshooting) on the Continue Docs

### Relevant environment info

```Markdown
- OS: Windows 11 Pro 64-Bit-Version (10.0, Build 22631)
- Continue: 0.0.56
- IDE: IntelliJ IDEA 2024.1.4 (Build #IU-241.18034.62)
- Model: gpt-4o
- config.json:
  
{
  "disableIndexing": true,
  "models": [
    {
      "title": "GPT-4o",
      "provider": "openai",
      "model": "gpt-4o",
      "systemMessage": "You are an expert software developer. You give helpful and concise responses.",
      "contextLength": 100000,
      "completionOptions": {
        "maxTokens": 4000
      }
    }
  ],
  "customCommands": [],
  "allowAnonymousTelemetry": false,
  "docs": []
}
```

### Description

In the past, I have sporadically noticed that the IDE freezes when using the inline edit functionality of Continue and can only be closed via the Task Manager. Some colleagues have also told me that they have had similar experiences.

I have now discovered a case today for which the behavior is reproducible. I have also been able to reproduce this on several end devices. I have attached the case and also a video of the misbehavior.

https://github.com/user-attachments/assets/bc2f55c5-1d84-4d25-ae48-580b7899da23

(The behavior does not occur under VSCode, I have validated this).

### To reproduce

1. open a Java File containing the following content:
```java
package de.test;

import com.fasterxml.jackson.annotation.JsonIgnore;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.context.SecurityContextHolder;

import java.util.Collection;
import java.util.Set;

public abstract class Model {
    private String name;
    private String label;

    @JsonIgnore
    private Set<String> authorities;

    public String getName() {
        return this.name;
    }

    public void setName(final String name) {
        this.name = name;
    }

    public String getLabel() {
        return this.label;
    }

    public void setLabel(final String label) {
        this.label = label;
    }

    public Set<String> getAuthorities() {
        return this.authorities;
    }

    public void setAuthorities(final Set<String> authorities) {
        this.authorities = authorities;
    }

    /**
     * @return true if the user is allowed to use the model
     */
    public boolean isPermittedForCurrentUser() {
        final Set<String> modelAuthorities = getAuthorities();
        if (modelAuthorities == null || modelAuthorities.isEmpty()) {
            return true;
        }
        final Collection<? extends GrantedAuthority> userAuthorities = SecurityContextHolder.getContext()
                .getAuthentication().getAuthorities();
        for (final GrantedAuthority authority : userAuthorities) {
            if (modelAuthorities.contains(authority.getAuthority())) {
                return true;
            }
        }
        return false;
    }
}
```

2. mark the whole code using CTRL+A
3. start inline edits using CTRL+I
4. prompt "add missing javadoc"
5. submit
6. the ide crashes and can only be closed using the task manager (the code in the target file becomes invalid in the process)

### Log output

```Shell
[info] Starting Continue core...
[2024-08-07T13:55:54] [info] Starting Continue core... 
[2024-08-07T13:55:54] Setup 
[2024-08-07T13:55:54] Core started 
[2024-08-07T13:55:54] Unknown context provider codebase 
[2024-08-07T13:56:11] [object Object] 
[2024-08-07T13:56:11] [object Object] 
[2024-08-07T13:56:11] [object Object] 
[2024-08-07T13:56:11] [object Object] 
[2024-08-07T13:56:11] [object Object] 
[2024-08-07T13:56:11] [object Object] 
[2024-08-07T13:56:11] [object Object] 
[2024-08-07T13:56:11] [object Object] 
[2024-08-07T13:56:11] [object Object] 
[2024-08-07T13:56:11] [object Object] 
[2024-08-07T13:56:11] [object Object] 
[2024-08-07T13:56:11] [object Object] 
[2024-08-07T13:56:11] [object Object] 
[2024-08-07T13:56:11] [object Object] 
[2024-08-07T13:56:11] [object Object] 
[2024-08-07T13:56:11] [object Object] 
[2024-08-07T13:56:11] [object Object] 
[2024-08-07T13:56:11] [object Object] 
[2024-08-07T13:56:11] [object Object] 
[2024-08-07T13:56:11] [object Object] 
[2024-08-07T13:56:11] [object Object] 
[2024-08-07T13:56:11] [object Object] 
[2024-08-07T13:56:11] [object Object] 
[2024-08-07T13:56:11] [object Object] 
[2024-08-07T13:56:11] [object Object] 
[2024-08-07T13:56:11] [object Object] 
[2024-08-07T13:56:11] [object Object] 
[2024-08-07T13:56:11] [object Object] 
[2024-08-07T13:56:11] [object Object] 
[2024-08-07T13:56:11] [object Object] 
[2024-08-07T13:56:11] [object Object] 
[2024-08-07T13:56:11] [object Object] 
[2024-08-07T13:56:11] [object Object] 
[2024-08-07T13:56:11] [object Object] 
[2024-08-07T13:56:11] [object Object] 
[2024-08-07T13:56:11] [object Object] 
[2024-08-07T13:56:11] [object Object] 
[2024-08-07T13:56:11] [object Object] 
[2024-08-07T13:56:11] [object Object] 
[2024-08-07T13:56:11] [object Object] 
[2024-08-07T13:56:11] [object Object] 
[2024-08-07T13:56:11] [object Object] 
[2024-08-07T13:56:12] [object Object] 
[2024-08-07T13:56:12] [object Object] 
[2024-08-07T13:56:12] [object Object] 
[2024-08-07T13:56:12] [object Object] 
[2024-08-07T13:56:12] [object Object] 
[2024-08-07T13:56:12] [object Object] 
[2024-08-07T13:56:12] [object Object] 
[2024-08-07T13:56:12] [object Object]
```

**Comment by Hakenadu** [comment]
when i saw the issue #1974, i switched back to the old UI and could actually see a difference. the code fragment i originally mentioned could be processed successfully several times, while the following slightly extended code fragment also causes the IDE to freeze again.

```java
package de.test;

import com.fasterxml.jackson.annotation.JsonIgnore;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.context.SecurityContextHolder;

import java.util.Collection;
import java.util.Set;

public abstract class Model {
    private String name;
    private String label;
    private String description; // New field
    private String type;        // New field
    private boolean active;     // New field

    @JsonIgnore
    private Set<String> authorities;

    public String getName() {
        return this.name;
    }

    public void setName(final String name) {
        this.name = name;
    }

    public String getLabel() {
        return this.label;
    }

    public void setLabel(final String label) {
        this.label = label;
    }

    public String getDescription() { // New getter
        return this.description;
    }

    public void setDescription(final String description) { // New setter
        this.description = description;
    }

    public String getType() { // New getter
        return this.type;
    }

    public void setType(final String type) { // New setter
        this.type = type;
    }

    public boolean isActive() { // New getter
        return this.active;
    }

    public void setActive(final boolean active) { // New setter
        this.active = active;
    }

    public Set<String> getAuthorities() {
        return this.authorities;
    }

    public void setAuthorities(final Set<String> authorities) {
        this.authorities = authorities;
    }

    /**
     * @return true if the user is allowed to use the model
     */
    public boolean isPermittedForCurrentUser() {
        final Set<String> modelAuthorities = getAuthorities();
        if (modelAuthorities == null || modelAuthorities.isEmpty()) {
            return true;
        }
        final Collection<? extends GrantedAuthority> userAuthorities = SecurityContextHolder.getContext()
                .getAuthentication().getAuthorities();
        for (final GrantedAuthority authority : userAuthorities) {
            if (modelAuthorities.contains(authority.getAuthority())) {
                return true;
            }
        }
        return false;
    }
}
```

### Old IDEA UI doesn't freeze with the (shorter) code fragment from the [original post](https://github.com/continuedev/continue/issues/1954#issue-2453558907)
https://github.com/user-attachments/assets/f3843765-2f6f-4d1f-ad44-601c9ebd24ef

### Old IDEA UI freezes with the slightly extended code fragment mentioned above
https://github.com/user-attachments/assets/abc4de31-2c7c-4970-8b7d-5ac4cfe6c4b5

**Comment by Hakenadu** [comment]
the error seems to be fixed with version 0.0.62

https://github.com/user-attachments/assets/4e64f73f-3186-4f9a-8fa1-75236d812d52

thanks!!
